online-shopping-website/src/Components/ProductDetails.js

```diff
@@ -0,0 +1,128 @@
+import Stack from '@mui/material/Stack';
+import AddIcon from '@mui/icons-material/Add';
+import RemoveIcon from '@mui/icons-material/Remove';
+import {useParams} from 'react-router-dom';
+import Products from '../TestValues.json';
+
+class ProductButtons extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            quantity: 1,
+            show: true,
+            max: 5,
+            min: 0
+        };
+    }
+
+    IncrementItem = () => {
+        if (this.state.quantity != 10) {
+            this.setState({
+                quantity: this.state.quantity + 1
+            });
+        }
+    }
+    DecreaseItem = () => {
+        if (this.state.quantity != 1) {
+            this.setState({quantity: this.state.quantity - 1});
+        }
+    }
+
+    UpdateValue = (e) => {
+        this.setState({quantity: e.target.value});
+    }
+
+    get quantity() {
+        return this.state.quantity;
+    }
+
+    render() {
+        return (
+            <div className="ProductDetails-QuantityButtonsContainer">
+                <h3>Quantity</h3>
+                <Stack className="ProductDetails-QuantityButtonsStack" direction="row" spacing={1}>
+                    <Button className="ProductDetails-QuantityButtons" variant="contained"
+                            disabled={this.state.quantity == 10}
+                            onClick={this.IncrementItem}>
+                        <AddIcon/>
+                    </Button>
+                    <input className="inputne" value={this.state.quantity} onChange={this.UpdateValue}/>
+                    <Button className="ProductDetails-QuantityButtons" variant="contained"
+                            disabled={this.state.quantity == 1}
+                            onClick={this.DecreaseItem}>
+                        <RemoveIcon/>
+                    </Button>
+                </Stack>
+                <h5 className="ProductDetails-ProductLimitText">Limit of 10 items per product in cart.</h5>
+                <Button className="ProductDetails-CartButton" variant="contained" endIcon={<AddShoppingCartIcon/>}>
+                    Add to cart
+                </Button>
+            </div>
+        );
+    }
+}
+
+export const ProductDetails = () => {
+    const productName = useParams();
+
+    const selectedProduct = {
+        brand: "",
+        description: "",
+        id: null,
+        image: "",
+        name: "",
+        price: null,
+        seller: ""
+    }
+
+    Products.products.map(product => {
+        if (productName.productName === product.name) {
+            selectedProduct.brand = product.brand;
+            selectedProduct.description = product.description;
+            selectedProduct.id = product.id;
+            selectedProduct.image = product.image;
+            selectedProduct.name = product.name;
+            selectedProduct.price = product.price.toFixed(2);
+            selectedProduct.seller = product.seller;
+        }
+    });
```

Comment:
`map()` expects a return for each element, to change each element in products. Would a `forEach()` make more sense here since you're not changing each element, but instead getting it?

Reply:
Leaving that for sprint 4 in refactoring.